src/valid/error.rs

```diff
@@ -33,58 +35,67 @@ impl<E: Display> Display for ValidationError<E> {
     }
 }
 
-impl<E> ValidationError<E> {
-    pub fn as_vec(&self) -> &Vec<Cause<E>> {
+impl<E: std::cmp::Eq + std::hash::Hash> ValidationError<E> {
+    pub fn as_vec(&self) -> &IndexSet<Cause<E>> {
```

Comment:
instead of this, we can have a separate method called `as_index_set`

Reply:
Having a `dedupe` operator on `ValidationError` is another option to remove duplicates.